### PR TITLE
NAS-3782 reimplemented declarative classes with attrs and cattrs

### PR DESCRIPTION
**Pull request by hkad98**
* classes uses attrs and cattrs which should make development easier in the future
* alias @attr.s was used because IntelliJ supports hinting for that when it supports new aliases such as define, the new aliases will be used

### CODE REVIEW

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

**Review by jaceksan — APPROVED** [review]

File: gooddata-sdk/type-requirements.txt [hunk 1]
```diff
@@ -2,3 +2,6 @@ mypy==0.910
 pydantic==1.8.2
 types-python-dateutil >= 2.5.3
 types-pyyaml~=6.0.0
+pyyaml==6.0
```
Comment (pcerny) [inline gooddata-sdk/type-requirements.txt:5]: Why is necessary to install pyyaml for mypy tests? `types-pyyaml` should be enough?

**Review by pcerny — CHANGES_REQUESTED** [review]

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

File: gooddata-sdk/type-requirements.txt (hunk 1 above)
Comment (hkad98) [reply gooddata-sdk/type-requirements.txt]: You are right. Copy paste error, thanks for spotting it.

**Review by pcerny — APPROVED** [review]

**Comment by yenkins** [comment]
### Sonar scan result


More detail, see in https://sonarqube-gate.intgdc.com/dashboard?id=gooddata-python-sdk-gate-PR103

To scan for vulnerabilities in dependencies and run unit tests (to get coverage report in sonar) please comment your PR with 'extended check sonar'.

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-python-sdk-sonarqube-scan-zuul-gate/95/ : SUCCESS in 59s
- https://checklist.intgdc.com/job/compliance/job/gooddata-python-sdk-fossa-scanning-zuul-el7/95/ : SUCCESS in 6m 28s (non-voting)
- https://checklist.intgdc.com/job/compliance/job/gooddata-python-sdk-merge-compliance-check/95/ : SUCCESS in 24s

**Pull request merged**